x-pack/plugins/lens/public/indexpattern_datasource/utils.tsx

```diff
@@ -101,3 +111,38 @@ export function fieldIsInvalid(column: IndexPatternColumn | undefined, indexPatt
   }
   return !!getInvalidFieldMessage(column, indexPattern)?.length;
 }
+
+export function getPrecisionErrorWarningMessages(
+  state: IndexPatternPrivateState,
+  { activeData }: FramePublicAPI
+) {
+  const warningMessages: React.ReactNode[] = [];
+
+  if (state && activeData) {
+    const hasPrecisionError = Object.values(activeData).some(
+      search.checkDatatableForPrecisionError
+    );
+
+    if (hasPrecisionError) {
+      warningMessages.push(
+        <FormattedMessage
+          key={`precision-error`}
```

Comment:
Why this static key value?

Reply:
@dej611  only one error of this type is available, besides, I looked below how warningMessages is used on other places and saw that we define a key for the rest of the elements.

like here: 
![image](https://user-images.githubusercontent.com/20072247/140073838-93c9b837-49fb-4452-99c2-bab170771ce7.png)

maybe we don't need to define that prop at all, but it was done in a common manner.

Reply:
In the case you are showing it is used to force a render on `columnId` change. In the case above it is forcing the component to never rerender - as it is static.

Reply:
np, checked that, all works fine after removing key